sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ForLoopIncrementSignCheck.cs

```diff
@@ -34,25 +34,206 @@ namespace SonarAnalyzer.Rules.CSharp
+                    var forNode = (ForStatementSyntax)c.Node;
+
+                    var conditionSyntax = forNode.Condition;
+                    if (conditionSyntax == null || !(conditionSyntax is BinaryExpressionSyntax binaryCondition))
                     {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, node.GetLocation()));
+                        return;
+                    }
+
+                    if (forNode.Incrementors.Count != 1)
+                    {
+                        return;
+                    }
+
+                    var incrementor = forNode.Incrementors[0];
+
+                    var incrementorData = GetIncrementData(incrementor);
+                    if (incrementorData.Operation == ArithmeticOperation.None)
+                    {
+                        return;
+                    }
+
+                    var condition = GetCondition(binaryCondition, incrementorData.IdentifierName);
+                    if (condition == Condition.None)
+                    {
+                        return;
+                    }
+
+                    if (incrementorData.Operation == ArithmeticOperation.Addition &&
+                        condition == Condition.Greater)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, forNode.Condition.GetLocation(), incrementorData.IdentifierName, "inc"));
+                    }
+                    else if (incrementorData.Operation == ArithmeticOperation.Substraction &&
+                             condition == Condition.Less)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, forNode.Condition.GetLocation(), incrementorData.IdentifierName, "dec"));
                     }
                 },
                 SyntaxKind.ForStatement);
         }
+
+        private IncrementData GetIncrementData(ExpressionSyntax incrementor)
+        {
+            var identifierName = string.Empty;
+            var opp = ArithmeticOperation.None;
+
+            var incrementorKind = incrementor.Kind();
+
+            switch (incrementorKind)
+            {
+                case SyntaxKind.PreDecrementExpression:
+                case SyntaxKind.PreIncrementExpression:
+                case SyntaxKind.PostDecrementExpression:
+                case SyntaxKind.PostIncrementExpression:
+                    if (GetOperandFromUnaryExpression(incrementor) is IdentifierNameSyntax operand)
+                    {
+                        identifierName = operand.Identifier.ValueText;
+                        opp =  incrementorToOperation.GetValueOrDefault(incrementorKind, ArithmeticOperation.None);
+                    }
+                    break;
+
+                case SyntaxKind.SubtractAssignmentExpression:
+                case SyntaxKind.AddAssignmentExpression:
+                    if (incrementor is AssignmentExpressionSyntax add && add.Left is IdentifierNameSyntax leftId &&
```

Comment:
nitpick: put the `add.Left is IdentifierNameSyntax leftId` on a new line (it's easier to follow if one line has one condition)

Reply:
Done